spec/api/comment_thread_spec.rb

```diff
@@ -592,7 +592,24 @@ def thread_result(id, params)
         comment.reload
         comment.endorsed.should == false
         comment.endorsement.should == nil
-        check_thread_result_json(nil, changed_thread, parse(last_response.body))
+        json_response = parse(last_response.body)
+        check_thread_result_json(nil, changed_thread, json_response)
```

Comment:
@wajeeha-khalid In your comment you mention the GET cases.  I imagine they should match.  It seems like it should be more clear where you would ever want check_thread_result_json with user of nil, as opposed to getting the user.

At a minimum, these 2 update tests should match.  In other words, we should get the user and sent it in here.